apps/web-giddh/src/app/settings/integration/setting.integration.component.ts

```diff
@@ -1631,6 +1659,349 @@ export class SettingIntegrationComponent implements OnInit, AfterViewInit {
+     */
+    public saveNewAccount(): void {
+        if (!this.createNewAccountForm?.invalid) {
+            this.createNewAccountForm.get('paymentAlerts')?.patchValue(this.paymentAlerts.filter(user => user !== this.selectAllRecords).map(user => user));
+
+            this.settingsIntegrationService.bankAccountRegistration(this.createNewAccountForm.value).pipe(take(1)).subscribe(response => {
+                if (response?.status === "success") {
+                    this.closeCreateNewAccountModal();
+                    if (response?.body?.message) {
+                        this.toasty.clearAllToaster();
+                        this.toasty.successToast(response?.body?.message);
+                    }
+                    this.getAllBankAccounts();
+                } else {
+                    this.toasty.clearAllToaster();
+                    this.toasty.errorToast(response?.message);
+                }
+            });
+        }
+    }
+
+    /**
+     * This will close the create new account modal
+     *
+     * @memberof SettingIntegrationComponent
+     */
+    public closeCreateNewAccountModal(): void {
+        this.createNewAccountModal?.hide();
+    }
+
+    /**
+     * This will open the create new account user modal
+     *
+     * @param {*} bankAccount
+     * @memberof SettingIntegrationComponent
+     */
+    public openCreateNewAccountUserModal(bankAccount: any): void {
+        this.activeBankAccount = bankAccount;
+
+        this.createNewAccountUserForm = this._fb.group({
+            accountUniqueName: [bankAccount?.account?.uniqueName, Validators.required],
+            accountNumber: [bankAccount?.iciciDetailsResource?.accountNumber, Validators.required],
+            userUniqueName: ['', Validators.required],
+            uniqueName: [bankAccount?.iciciDetailsResource?.uniqueName, Validators.required],
+            loginId: ['', Validators.required],
+            maxAmount: [''],
+            duration: ['']
+        });
+
+        this.createNewAccountUserModal?.show();
+    }
+
+    /**
+     * This will close the create new account user modal
+     *
+     * @memberof SettingIntegrationComponent
+     */
+    public closeCreateNewAccountUserModal(): void {
+        this.createNewAccountUserModal?.hide();
+    }
+
+    /**
+     * This will get all connected bank accounts
+     *
+     * @memberof SettingIntegrationComponent
+     */
+    public getAllBankAccounts(): void {
+        this.isLoading = true;
+        this.connectedBankAccounts = [];
+
+        this.settingsIntegrationService.getAllBankAccounts().pipe(take(1)).subscribe(response => {
+            this.isLoading = false;
+            if (response?.body) {
+                this.connectedBankAccounts = response.body;
+
+                this.connectedBankAccounts.forEach(bankAccount => {
+                    if(bankAccount?.iciciDetailsResource?.payor?.length > 0) {
+                        bankAccount?.iciciDetailsResource?.payor.forEach(payor => {
```

Comment:
Avoid API calls in for loop

Reply:
Please ignore as it's approved by Shubhendra sir